vms/platformvm/validators/manager.go

```diff
@@ -98,17 +99,19 @@ type State interface {
 func NewManager(
 	log logging.Logger,
 	cfg config.Config,
+	stateLock sync.Locker,
 	state State,
 	metrics metrics.Metrics,
 	clk *mockable.Clock,
 ) Manager {
 	return &manager{
-		log:     log,
-		cfg:     cfg,
-		state:   state,
-		metrics: metrics,
-		clk:     clk,
-		caches:  make(map[ids.ID]cache.Cacher[uint64, map[ids.NodeID]*validators.GetValidatorOutput]),
+		log:       log,
+		cfg:       cfg,
+		stateLock: stateLock,
```

Comment:
Do we need `stateLock` in `makePrimaryNetworkValidatorSet` and `makeSubnetValidatorSet`? We access `state` in those methods

Reply:
`getCurrentPrimaryValidatorSet` and `getCurrentValidatorSets` were separated into helpers to explicitly delineate the functionality that needs the lock.